Cloud SQL IAM service account authentication failed for user

### Bug Description

An otherwise valid configuration on occasion will result in `Cloud SQL IAM service account authentication failed for user`.

The backend will log: `Failed to validate access token` (as opposed to access token expired).

While customers have seen this error in Cloud Run (possibly suggesting a CPU-throttling issue with the background refresh), it also appears on GKE. 

Updating to the latest version has not resolved these occasional errors.

### Example code (or command)

_No response_

### Stacktrace

_No response_

### Steps to reproduce?

1. Deploy an app that logs in with Auto IAM AuthN
2. Wait awhile
3. Observe occasional failures

### Environment

1. OS type and version: Linux Container
2. Java SDK version: ?
3. Cloud SQL Java Socket Factory version: v1.10.0


### Additional Details

_No response_

Observed this issue twice since January on three persistent cloud run nodes - cpu throttling disabled.  Hikari pool size 10.

Feb 10th - v1.8.3 - project A - single node failure
March 3rd - v1.10.0 - project B - single node failure

Nodes recovered on their own after 12 and 17 minutes respectively.  No relevant problem indicators in metrics or logs around the time.

We have another service with a similar configuration that has not yet encountered the issue running v1.8.3 with 4 persistent nodes across 2 projects.

confirmed problem image based on `eclipse-temurin:11.0.17_8-jre-focal`, yet unaffected image based on `eclipse-temurin:11.0.17_8-jre-alpine`

service:
```
java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after (xxxx)ms.
   (...)
Caused by: org.postgresql.util.PSQLException: FATAL: Cloud SQL IAM service account authentication failed for user "xxxx"
```

postgres:
```
FATAL:  Cloud SQL IAM service account authentication failed for user "xxxx""
```

> 10:45 ET - "timeouts" begin

I assume that means you're seeing this:

```
java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after (xxxx)ms.
   (...)
Caused by: org.postgresql.util.PSQLException: FATAL: Cloud SQL IAM service account authentication failed for user "xxxx"
```

Yes?

--- a/core/src/main/java/com/google/cloud/sql/core/SqlAdminApiFetcher.java
+++ b/core/src/main/java/com/google/cloud/sql/core/SqlAdminApiFetcher.java
@@ -358,10 +358,11 @@ public class SqlAdminApiFetcher {
         DateTimeFormatter formatter = DateTimeFormatter.ISO_INSTANT.withZone(ZoneId.of("UTC"));
         String nowFormat = formatter.format(now);
         String expirationFormat = formatter.format(expirationTime);
-        String errorMessage = "Access Token expiration time is in the past. Now = "
-            + nowFormat
-            + " Expiration = "
-            + expirationFormat;
+        String errorMessage =
+            "Access Token expiration time is in the past. Now = "
+                + nowFormat
+                + " Expiration = "
+                + expirationFormat;
         logger.warning(errorMessage);
         throw new RuntimeException(errorMessage);
       }

--- a/core/src/main/java/com/google/cloud/sql/core/SqlAdminApiFetcher.java
+++ b/core/src/main/java/com/google/cloud/sql/core/SqlAdminApiFetcher.java
@@ -358,17 +358,20 @@ public class SqlAdminApiFetcher {
         DateTimeFormatter formatter = DateTimeFormatter.ISO_INSTANT.withZone(ZoneId.of("UTC"));
         String nowFormat = formatter.format(now);
         String expirationFormat = formatter.format(expirationTime);
-        throw new RuntimeException(
-            "Access Token expiration time is in the past. Now = "
-                + nowFormat
-                + " Expiration = "
-                + expirationFormat);
+        String errorMessage = "Access Token expiration time is in the past. Now = "
+            + nowFormat
+            + " Expiration = "
+            + expirationFormat;
+        logger.warning(errorMessage);
+        throw new RuntimeException(errorMessage);
       }
     }
 
     // Is the token empty?
     if (tokenValue.length() == 0) {
-      throw new RuntimeException("Access Token has length of zero");
+      String errorMessage = "Access Token has length of zero";
+      logger.warning(errorMessage);
+      throw new RuntimeException(errorMessage);
     }
   }
 

